src/crypto/chacha20.cpp

```diff
@@ -100,6 +102,10 @@ void ChaCha20::Output(unsigned char* c, size_t bytes)
 
     for (;;) {
         if (bytes < 64) {
+            if (m != nullptr) {
+                for (i = 0;i < bytes;++i) tmp[i] = m[i];
```

Comment:
A comment would be helpful here: "If m has fewer than 64 bytes available, copy m to tmp and read from tmp instead"

Reply:
Added.